package/lib/src/beam_interceptor_pop_scope.dart

```diff
@@ -0,0 +1,71 @@
+import 'package:beamer/beamer.dart';
+import 'package:flutter/widgets.dart';
+
+/// This works like [PopScope], but with beam-interceptors.
+///
+/// See [BeamInterceptor] for more information.
+///
+/// If any of the interceptors return true, the pop will not be invoked.
+///
+/// This works on Navigator.maybePop as well as all the Beamer.beamTo, Beamer.beamToNamed, Beamer.beamBack... operations.
+///
+/// ```dart
+/// BeamInterceptorPopScope(
+///   interceptors: [BeamInterceptor(...), ...],
+///   child: Center(
+///     child: ElevatedButton(
+///       child: const Text('Go back'),
+///       onPressed: () => Beamer.of(context).beamToNamed('/'),
+///     ),
+///   ),
+/// );
+/// ```
+class BeamInterceptorPopScope extends StatefulWidget {
+  const BeamInterceptorPopScope({
+    required this.child,
+    required this.interceptors,
+    this.beamerDelegate,
+    super.key,
+  });
+
+  final Widget child;
+
+  /// The interceptors to check when a maybePop, or beamTo, beamToNamed, beamBack... is triggered.
+  final List<BeamInterceptor> interceptors;
+
+  /// The beamerDelegate to apply the interceptors to.
```

Comment:
Let's change this to this, to have the class reference
> /// The [BeamerDelegate] to apply the interceptors to.

Reply:
✅